var/spack/repos/builtin/packages/py-pycompadre/package.py

```diff
@@ -35,19 +38,30 @@ class PyPycompadre(PythonPackage):
         description="Debugging level 0) release 1) debug 2) extreme debugging",
     )
 
-    depends_on("cmake@3.10.0:", type="build")
-    depends_on("python@3.4:", type=("build", "link", "run"))
-    depends_on("py-pip", type=("build", "link", "run"))
+    depends_on("cmake@3.13:", type="build", when="@:1.4")
+    depends_on("cmake@3.16:", type="build", when="@1.5:")
+
+    depends_on("kokkos-kernels@3.3.01:4", when="@:1.5")
+    depends_on("kokkos-kernels@4:", when="@1.6:")
+
+    depends_on("python@3.6:", type=("build", "link", "run"))
```

Comment:
FYI. `python@3.4:` is required for `1.5.0` per `setup.py`s with `@3.6:` starting with `1.6.0`.

Reply:
This is true. However `python@3.4:3.5` are far past end of life and nearly were when those requirements were published. While setup.py from those version may permit installation, I would not recommend using any Python below 3.6.0 from a security standpoint. Is the preferred approach to match the setup.py, even when what is in there is no longer suggested? At the time of those releases, Python 3.6 was primarily used for CI testing.

Reply:
That said, there are some projects or sites that may still rely on older versions of python (depending on the sizes and ages of their code bases, for example) that may still only have older versions of python available. So my preference is to match; however, I'm sure others (or most?) agree with you. 😁